pages/admin/settings.js

```diff
@@ -1,19 +1,169 @@
+        formRef.current.resetForm({ values: {} });
+        setStaticLink('');
+      } catch (ex) {
+        console.error(ex);
+      } finally {
+        dispatch({ type: DELETING, payload: false });
+      }
+    })();
+  }, []);
+
+  console.log(state.values)
+
+  const onSubmit = (evt) => {
+    evt.preventDefault();
+    formRef.current.handleSubmit();
+  };
+
+  const getS3Keys = (files) => {
+    if(!files) return
+
+    const fileKeys = Object.keys(files);
+    const s3Keys = fileKeys.map((file) => state.values[file].key);
+    
+    return s3Keys;
+  };
+
+  const formikParams = {
+    innerRef: formRef,
+    initialValues: state.values,
+    onSubmit: (values) => {
+      (async () => {
+        try {
+          dispatch({ type: SAVING });
+          const isFile = Object.entries(values)[0][1] instanceof File;
+          const isCreateMode = !isValueExists && isFile;
+       
+          const filesToUpload = !isCreateMode && getAllFiles(values);
+          const s3Keys = getS3Keys(filesToUpload);   
+          const newValues = isCreateMode ? values : {...values, toDeleteRaw: s3Keys}
+
+          const formattedEntries = convertToFormData(newValues);
+              
+          const response = await slsFetch(`/api/resources${isCreateMode ? '' : '/mainlogo'}`, {
+            method: `${isCreateMode ? "POST" : "PUT"}`,
+            body: formattedEntries,
+          });
+          const { data } = await response.json()
+
+          const newData = setInitialValues(data);
+          dispatch({ type: SET_VALUES, payload: newData });
+          formRef.current.resetForm({ values: newData });
+        } catch (ex) {
+          console.error(ex);
+        } finally {
+          dispatch({ type: CLEANUP });
+        }
+      })();
+    },
+  };
 
-export default function Settings({cache}) {
   return (
     <CacheContext.Provider value={cache}>
-      <InnerLayout>
+      <InnerSingleLayout>
         <div>
           <div className="row">
-            <div className="col-xl-8 col-lg-8 col-md-8">
-              <h3 className="mt-5"> Settings </h3>
+            <div className="col-12">
+              <div className="row mt-5">
+                <div className="col-12 col-md-10">
+                  <h3>Settings</h3>
+                </div>
+                <div className="col-12 col-md-2 float-right">
+                  <AppButtonLg
+                    title={state.isSaving ? "Saving" : "Save"}
+                    icon={state.isSaving ? <AppButtonSpinner /> : <FaCheck />}
+                    onClick={onSubmit}
+                    isDisabled={state.isLoading || state.isSaving || state.isDeleting}
+                  />
+                </div>
+              </div>
+              {!state.isLoading && (
+                <Formik {...formikParams}>
+                  {() => (
+                    <Form>
+                      <UploadRenderer
+                        accept="image/png, image/gif, image/jpeg, image/jfif"
```

Comment:
The string `image/png, image/gif, image/jpeg, image/jfif` appeared multiple times in this pull request. Maybe refactor the constant in a central file and export it instead, so that when the time comes when we restrict or add new data types, we modify it in one place?